stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewHighlighter.java

```diff
@@ -102,15 +105,48 @@ private void highlightViewOnUiThread() {
       }
 
       if (mHighlightedView != null) {
-        mHighlightedView.getOverlay().remove(mHighlightDrawable);
+        for (ColorDrawable drawable : mHighlightDrawables) {
+          mHighlightedView.getOverlay().remove(drawable);
+        }
       }
 
-      if (viewToHighlight != null) {
-        mHighlightDrawable.setColor(mContentColor.get());
-        mHighlightDrawable.setBounds(0, 0, viewToHighlight.getWidth(), viewToHighlight.getHeight());
-        viewToHighlight.getOverlay().add(mHighlightDrawable);
+      if (viewToHighlight != null &&
+          viewToHighlight.getLayoutParams() instanceof ViewGroup.MarginLayoutParams) {
+
+        mHighlightDrawables = newViewHighlightDrawablesArray(viewToHighlight);
+
+        for (ColorDrawable drawable : mHighlightDrawables) {
+          viewToHighlight.getOverlay().add(drawable);
+        }
+
         mHighlightedView = viewToHighlight;
       }
     }
+
+    private ColorDrawable[] newViewHighlightDrawablesArray(View viewToHighlight) {
```

Comment:
Doesn't seem like we need to create this array each time, but rather to take the drawables as input in some fashion (maybe using a container class) and update their bounds?  Allocating data in UI hot paths (even for a debugging tool) is discouraged :)